src/main/java/net/dv8tion/jda/api/interactions/callbacks/IPremiumRequiredReplyCallback.java

```diff
@@ -16,22 +16,31 @@
 
 package net.dv8tion.jda.api.interactions.callbacks;
 
-import net.dv8tion.jda.api.requests.restaction.interactions.PremiumRequiredCallbackAction;
 import net.dv8tion.jda.api.entities.Entitlement;
+import net.dv8tion.jda.api.entities.SkuSnowflake;
+import net.dv8tion.jda.api.interactions.components.buttons.Button;
+import net.dv8tion.jda.api.requests.restaction.interactions.PremiumRequiredCallbackAction;
 
 import javax.annotation.CheckReturnValue;
 import javax.annotation.Nonnull;
 
 /**
+ * @deprecated Replaced with {@link Button#premium(SkuSnowflake, String)}
+ *
  * Replies with an in-built client message stating that an {@link Entitlement Entitlement} is required.
  *
  * <p>Replying with {@link #replyWithPremiumRequired()} will automatically acknowledge this interaction.
  *
  * <p><b>Note:</b>This interaction requires <a href="https://discord.com/developers/docs/monetization/overview" target="_blank">monetization</a> to be enabled.
  */
+@Deprecated
 public interface IPremiumRequiredReplyCallback extends IDeferrableCallback
 {
+    /**
+     * @deprecated Replaced with {@link Button#premium(SkuSnowflake, String)}
```

Comment:
Link to the [changelog](https://discord.com/developers/docs/change-log#premium-apps-new-premium-button-style-deep-linking-url-schemes) here to further elaborate